From ab2df2f35f52c16689f29e50f647b8d3405fd86b Mon Sep 17 00:00:00 2001
From: Steve Zesch <stevezesch2@gmail.com>
Date: Sat, 4 Aug 2012 19:06:08 -0400
Subject: [PATCH] Micro-optimization.

do_multimedia_player_action did nothing more than call
msd_media_player_key_pressed, which was pointless overhead.
The do_multimedia_player_action calls have now been replaced
with calls to msd_media_player_key_pressed.
---
 plugins/media-keys/msd-media-keys-manager.c |   17 +++++------------
 1 file changed, 5 insertions(+), 12 deletions(-)

diff --git a/plugins/media-keys/msd-media-keys-manager.c b/plugins/media-keys/msd-media-keys-manager.c
index d75c031..f65596d 100644
--- a/plugins/media-keys/msd-media-keys-manager.c
+++ b/plugins/media-keys/msd-media-keys-manager.c
@@ -980,13 +980,6 @@ msd_media_player_key_pressed (MsdMediaKeysManager *manager,
 }
 
 static gboolean
-do_multimedia_player_action (MsdMediaKeysManager *manager,
-                             const char          *key)
-{
-        return msd_media_player_key_pressed (manager, key);
-}
-
-static gboolean
 do_action (MsdMediaKeysManager *manager,
            int                  type)
 {
@@ -1059,19 +1052,19 @@ do_action (MsdMediaKeysManager *manager,
                 g_free (cmd);
                 break;
         case PLAY_KEY:
-                return do_multimedia_player_action (manager, "Play");
+                return msd_media_player_key_pressed (manager, "Play");
                 break;
         case PAUSE_KEY:
-                return do_multimedia_player_action (manager, "Pause");
+                return msd_media_player_key_pressed (manager, "Pause");
                 break;
         case STOP_KEY:
-                return do_multimedia_player_action (manager, "Stop");
+                return msd_media_player_key_pressed (manager, "Stop");
                 break;
         case PREVIOUS_KEY:
-                return do_multimedia_player_action (manager, "Previous");
+                return msd_media_player_key_pressed (manager, "Previous");
                 break;
         case NEXT_KEY:
-                return do_multimedia_player_action (manager, "Next");
+                return msd_media_player_key_pressed (manager, "Next");
                 break;
         default:
                 g_assert_not_reached ();
-- 
1.7.9.5

